### PR TITLE
Refine error handling for JVM dependencies that fail to resolve.

### PR DESCRIPTION
**Pull request by olafurpg**
- Use `coursier resolve` instead of `coursier complete` to determine the
  existence of a dependency. The complete method is only intended to
  power tab completions, it can still have false positives/negatives
  when determining whether a dependency exists.
- Don't fail fast if a single version fails to resolve. Now, the repo
  gets created as long as at least one version resolves successfully.
  Versions that fail to resolve get logged as warnings instead (unless
  all versions fail to resolve, which is a hard error).
- Use the `--quit` flag when resolving dependencies to reduce logs noise.

### CODE REVIEW

File: cmd/gitserver/server/vcs_syncer_jvm_packages.go [hunk 1]
```diff
@@ -95,9 +95,10 @@ func (s *JVMPackagesSyncer) CloneCommand(ctx context.Context, remoteURL *vcs.URL
 	return exec.CommandContext(ctx, "git", "--version"), nil
 }
 
-// Fetch does nothing for Maven packages because they are immutable and cannot be updated after publishing.
+// Fetch add git tags for newly added dependency versions and removes git tags
```
Comment (Strum355) [inline cmd/gitserver/server/vcs_syncer_jvm_packages.go:98]:
```suggestion
// Fetch adds git tags for newly added dependency versions and removes git tags
```

File: internal/repos/jvm_packages.go [hunk 2]
```diff
@@ -70,33 +72,45 @@ func (s *JVMPackagesSource) GetRepo(ctx context.Context, artifactPath string) (*
...
-type mavenArtifactNotFound struct {
-	dependency reposource.MavenDependency
+type jvmDependencyNotFound struct {
+	dependencies []reposource.MavenDependency
 }
 
-func (mavenArtifactNotFound) NotFound() bool {
+func (jvmDependencyNotFound) NotFound() bool {
 	return true
 }
```
Comment (Strum355) [inline internal/repos/jvm_packages.go:110]:
```suggestion
```

[Doesnt seem to be used](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d15bb5e510c8f32d775cb53244fddb72fbf2cab3/-/blob/internal/repos/jvm_packages.go?L108:30#tab=references)

**Review by Strum355 — APPROVED** [review]

File: cmd/gitserver/server/vcs_syncer_jvm_packages.go (hunk 1 above)
Comment (olafurpg) [reply cmd/gitserver/server/vcs_syncer_jvm_packages.go]: Good catch, done.

File: internal/repos/jvm_packages.go (hunk 2 above)
Comment (olafurpg) [reply internal/repos/jvm_packages.go]: Nice!

File: internal/extsvc/jvmpackages/coursier/coursier.go [hunk 3]
```diff
@@ -62,29 +59,29 @@ func FetchByteCode(ctx context.Context, config *schema.JVMPackagesConnection, de
...
+	_, err := runCoursierCommand(
 		ctx,
 		config,
-		"complete",
+		"resolve",
+		"--quiet", "--quiet",
 		dependency.CoursierSyntax(),
 	)
-	return len(versions) > 0, err
+	return err != nil
```
Comment (Strum355) [inline internal/extsvc/jvmpackages/coursier/coursier.go:84]:
```suggestion
	return err == nil
```

This caught me out big time while working on https://github.com/sourcegraph/sourcegraph/pull/22984 :^)

Comment (olafurpg) [reply internal/extsvc/jvmpackages/coursier/coursier.go:84]: Woah, nice catch. My bad 🙈

Comment (olafurpg) [reply internal/extsvc/jvmpackages/coursier/coursier.go]: `err != nil` is already built into muscle memory it's weird to write `err == nil`

**Comment by olafurpg** [comment]
The test is failing with a reasonable error. I won't be able to fix it today but I'll look into it early tomorrow

**Pull request merged**